ignite3/src/main/java/site/ycsb/db/ignite3/IgniteAbstractClient.java

```diff
@@ -140,86 +145,106 @@ public abstract class IgniteAbstractClient extends DB {
   protected static String partitions;
 
   /**
-   * Initialize any state for this DB. Called once per DB instance; there is one
-   * DB instance per client thread.
+   * Set IgniteServer instance to work with.
+   *
+   * @param igniteSrv Ignite.
    */
+  public static void setIgniteServer(IgniteServer igniteSrv) {
+    igniteServer = igniteSrv;
+    ignite = igniteServer.api();
+    externalIgnite = true;
+  }
+
+  /** {@inheritDoc} */
   @Override
   public void init() throws DBException {
     INIT_COUNT.incrementAndGet();
 
     synchronized (IgniteAbstractClient.class) {
-      if (ignite != null) {
+      if (initCompleted) {
```

Comment:
if another thread do init in parallel - it gets not initialized client. Let's use some synchronizes here, like:
```
// IgniteClient initialization timeout, seconds
private static final int INIT_TIMEOUT = 10; 
AtomicBoolean initCompleted = new AtomicBoolean(false);
CountDownLatch initLatch = new CountDownLatch(1);

public void init() throws DBException {
    if (initCompleted.compareAndSet(false, true) {
        initProperties(getProperties());
        initIgniteClientNode(useEmbeddedIgnite);
        initLatch.countDown();
    } else {
        initLatch.await(INIT_TIMEOUT, TimeUnit.SECONDS);
    }
}
```

Reply:
> it gets not initialized client

client is static, no need to init it twice

Reply:
Find out on call